File: plugins/javascript-kit/testdata/good/src/ok.js
'use strict';

const { readFile } = require('fs/promises');

async function readOk(path) {
  return readFile(path, 'utf8');
}

function mergeOk(body) {
  const out = Object.assign(Object.create(null), body);
  return out;
}

/* Named boundary docs; intentional legacy uses js-rg-allow on the smell line. */
function documentedLegacy(path) {
  const fs = require('fs');
  return fs.statSync(path); // js-rg-allow: boot-time view cache probe before async path binds
}

module.exports = { readOk, mergeOk, documentedLegacy };

File: plugins/cpp-kit/testdata/good/ok.cpp
#include <memory>
#include <cstdint>

std::unique_ptr<int> make_int(int v) {
  return std::make_unique<int>(v);
}

std::int64_t widen(int x) {
  return static_cast<std::int64_t>(x);
}

/* Named boundary docs; intentional legacy uses cpp-rg-allow on the smell line. */
int *documented_legacy(int n) {
  return new int[n]; /* cpp-rg-allow: sized by caller contract; prefer unique_ptr */
}
